string user_message
float64 sent_time